source/common/tls/utility.cc

```diff
@@ -226,6 +226,36 @@ std::string Utility::generalNameAsString(const GENERAL_NAME* general_name) {
     }
     break;
   }
+  case GEN_OTHERNAME:
+    ASN1_TYPE* value = general_name->d.otherName->value;
+    if (value == nullptr) {
+      break;
+    }
+    switch (value->type) {
+    case V_ASN1_NULL:
+      break;
+    case V_ASN1_BOOLEAN:
+      san = value->value.boolean ? "true" : "false";
+      break;
+    case V_ASN1_ENUMERATED:
+      san = std::to_string(ASN1_ENUMERATED_get(value->value.enumerated));
+      break;
+    case V_ASN1_INTEGER:
+      san = std::to_string(ASN1_INTEGER_get(value->value.integer));
+      break;
+    case V_ASN1_OBJECT: {
+      char tmp_obj[256]; // OID Max length
+      if (OBJ_obj2txt(tmp_obj, 256, value->value.object, 1) < 0) {
+        break;
+      }
+      san.assign(tmp_obj);
+      break;
```

Comment:
We can doc those in the API doc about those type will be convert as string.

Reply:
Added the failure test based on value and also updated docs. Please lmk if any changes are required